### PR TITLE
Add recompletion reset via a date

### PR DESCRIPTION
**Pull request by keevan**
The initial schedule/interval is currently set to a year.

image

### CODE REVIEW

File: classes/task/check_recompletion.php [hunk 1]
```diff
@@ -42,6 +42,51 @@ public function get_name() {
...
+                       coalesce(r3.value, '0') as nextresettime
+                  FROM {course_completions} cc
+                  JOIN {local_recompletion_config} r2 ON r2.course = cc.course
+                                                     AND r2.name = 'recompletiontype' AND r2.value = 'schedule'
+             LEFT JOIN {local_recompletion_config} r3 ON r3.course = cc.course AND r3.name = 'nextresettime'
+                  JOIN {local_recompletion_config} r4 ON r4.course = cc.course AND r4.name = 'recompletionschedule'
+                  JOIN {course} c ON c.id = cc.course
+                 WHERE c.enablecompletion = ".COMPLETION_ENABLED."
+                   AND cc.timecompleted > 0
+                   AND 1 = 1 LIMIT 3";
```
Comment (danmarsden) [inline classes/task/check_recompletion.php]: LImit 3? - debugging left in place accidentally?

**Comment by danmarsden** [comment]
nice work! - I like this a lot better than the cron format - just one comment on the added sql which looks like some accidental debugging left in place - but otherwise happy for this to land.

thanks!

File: recompletion.php [hunk 2]
```diff
@@ -102,10 +116,14 @@
         } else {
             if ($name == 'recompletionemailsubject' || $name == 'recompletionemailbody') {
                 $value = '';
+            } else if ($name === 'nextresettime') {
```
Comment (dmitriim) [inline recompletion.php]: as long as you are refactoring some part of the code, maybe you can rewrite this multiple if else block using switch?

Comment (keevan) [reply recompletion.php]: Thanks Dima, I'll see if there's a better way to clean up this section. Appreciate the feedback.

**Comment by danmarsden** [comment]
thanks @keevan - nice to have this land! :-)

**Pull request merged**